scaling/provider/aws.go

```diff
@@ -0,0 +1,118 @@
+package provider
+
+import (
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+type AWSConfig struct {
+	GuestFramework *framework.Guest
+	HostFramework  *framework.Host
+	Logger         micrologger.Logger
+
+	ClusterID string
+}
+
+type AWS struct {
+	guestFramework *framework.Guest
+	hostFramework  *framework.Host
+	logger         micrologger.Logger
+
+	clusterID string
+}
+
+func NewAWS(config AWSConfig) (*AWS, error) {
+	if config.GuestFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.GuestFramework must not be empty", config)
+	}
+	if config.HostFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HostFramework must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.ClusterID == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClusterID must not be empty", config)
+	}
+
+	a := &AWS{
+		guestFramework: config.GuestFramework,
+		hostFramework:  config.HostFramework,
+		logger:         config.Logger,
+
+		clusterID: config.ClusterID,
+	}
+
+	return a, nil
+}
+
+func (a *AWS) AddWorker() error {
+	customObject, err := a.hostFramework.AWSCluster(a.clusterID)
```

Comment:
I am not sure about the dependency graph here and how we should structure our libs. Do you think it makes sense to implement the provider specifics here instead of `e2e-harness`? I am not 100% sure about this but it kind of makes sense because we even have the provider naming here now to differentiate between provider specific differences across e2e tests. Maybe these primitives are better placed here and we keep the setup magic in `e2e-harness` and the test specifics in `e2etests`.

Reply:
Yes, makes sense to me to extract here specifics about setup/tests :+1:

Reply:
Ok cool. Lets see how this goes in the future. For now I moved some functionality of `e2e-harness` about working with the provider CRs to `e2etests`. Also see https://github.com/giantswarm/e2e-harness/pull/82.